p2p/server.go

```diff
@@ -833,6 +851,10 @@ func (srv *Server) run(dialstate dialer) {
 				p.RemovePurpose(purpose)
 			}
 		}
+		if ch == nil {
```

Comment:
I know all this changes come from master, but: 

nit: i think `ch == nil` is a convoluted way to asking: `HasNoPurpose && not in peers[id]`. I would try to make it more explicit of what `ch == nil` actually means (change it to something else).

In fact, i would make `ch` and `sub` local variables to the goroutine, and thus reduce their scope. No need to have them as global to the whole function.

Also, i now realize that when removing a static we subscribe to **all events from all peers**; probably there's not easier way of doing this, but seems like an overkill.

Reply:
Sounds good, I'll move them and add a boolean called `disconnecting` that can then be used in the `if`

As far as the subscription, it does look like there is no way to subscribe to only certain events, so there's not much else we could do.